File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/QNSRespondBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.QNSRespondBO;
import com.purefun.fstp.core.bo.pro.QNSRespondBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class QNSRespondBO_OTW implements ICommom_OTW {
    QNSRespondBO_PRO.QNSRespondBO.Builder builder = null;
    QNSRespondBO bo = null;

    public QNSRespondBO_OTW() {
        builder = QNSRespondBO_PRO.QNSRespondBO.newBuilder();
        bo= new QNSRespondBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public QNSRespondBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = QNSRespondBO_PRO.QNSRespondBO.newBuilder();
        bo= new QNSRespondBO();
        QNSRespondBO_PRO.QNSRespondBO receive = QNSRespondBO_PRO.QNSRespondBO.parseFrom(message);
        setServername(receive.getServername());
        setRespond(receive.getRespond());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public QNSRespondBO_OTW(QNSRespondBO bofrom){
        builder = QNSRespondBO_PRO.QNSRespondBO.newBuilder();
        bo= new QNSRespondBO();
        setServername(bofrom.servername);
        setRespond(bofrom.respond);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public QNSRespondBO getBo() { 
        return bo;
    }

    public java.lang.String getServername() {
        return builder.getServername();
    }

    public void setServername(java.lang.String servername) {
        bo.servername = servername;
        builder.setServername(servername);
    }

    public java.lang.String getRespond() {
        return builder.getRespond();
    }

    public void setRespond(java.lang.String respond) {
        bo.respond = respond;
        builder.setRespond(respond);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "QNSRespondBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "servername = " + getServername() +"," +
            "respond = " + getRespond() +"," +
         "]";
    }
}

File: fstp-core-bo/python/com/purefun/fstp/core/bo/otw/QNSRespondBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import QNSRespondBO_pb2
from com.purefun.fstp.core.bo.model.QNSRespondBO import QNSRespondBO

class QNSRespondBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = QNSRespondBO_pb2.QNSRespondBO()
        self._bo = QNSRespondBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.servername = self._bo.servername
        self._bo_pro.respond = self._bo.respond
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.servername = self._bo_pro.servername
        self._bo.respond = self._bo_pro.respond
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getServername(self):
        return self._bo.servername

    def setServername(self, servername):
        self._bo.servername = servername
        self._bo_pro.servername = servername

    def getRespond(self):
        return self._bo.respond

    def setRespond(self, respond):
        self._bo.respond = respond
        self._bo_pro.respond = respond

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "QNSRespondBO_OTW ["+"servername = " + self.getServername() +"," +"respond = " + self.getRespond() +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
